### ISSUE TITLE
Image resize enabled doesn't work

### ISSUE THREAD

**Issue by RuchitaCrawl** [ckeditor/ckeditor5#7272]
Hey!! Thank you for this awesome package.

I m trying to enable resize image in decoupled editor. But it thrown me the duplicate module error.
when i was trying to add resizeimage package in my build it don't shown me tool bar.

can u please tell me toolbar configurations or enable resize image feature properly.

my configurations is::

  

```
    import DecoupledEditor from '@ckeditor/ckeditor5-build-decoupled-document'
    import ImageResize from '@ckeditor/ckeditor5-image/src/imageresize'
```

plz note that above configuration give me an duplication of module error run time.
 import CKEditor from '@ckeditor/ckeditor5-vue';
   
  ```
    import DecoupledEditor from '@ckeditor/ckeditor5-editor-decoupled/src/decouplededitor';
    import BoldPlugin from '@ckeditor/ckeditor5-basic-styles/src/bold';
    import ItalicPlugin from '@ckeditor/ckeditor5-basic-styles/src/italic';
    import ParagraphPlugin from '@ckeditor/ckeditor5-paragraph/src/paragraph';

    components:{
            ckeditor: CKEditor.component,
        },

    data() {
            return {
                editor: DecoupledEditor,
                editorData: '',
                editorConfig: {
                    plugins: [
                        EssentialsPlugin,
                        BoldPlugin,
                        ItalicPlugin,
                        ParagraphPlugin,
                        // ImageResize
                    ],
                    ckfinder: {
                        // eslint-disable-next-line max-len
                      
                    uploadUrl:
'https://cksource.com/weuy2g4ryt278ywiue/core/connector/php/connector.php?command=QuickUpload&type=Files&responseType=json'
                    },
                    toolbar: {
                        items: [ 'bold', 'italic',  ],
                    },
                    image: {
                        resize: true,
                         toolbar: [ 'imageTextAlternative', '|', 'imageStyle:alignLeft', 'imageStyle:full', 'imageStyle:alignRight'],
                        styles: [
                            'full',
                            'alignLeft',
                            'alignRight',
                        ],

                    },
                },
        },
```

**Comment by FilipTokarski** [comment]
Hi, thanks for the report. The `duplicated modules error` might be caused by a few things, but most often it occurs when ckeditor plugins have different versions. So in the first place, please check if all the ckeditor plugins you installed have the same version ( latest is v19 ). If that's not the case, I'll need more information and detailed steps to reproduce this issue.

**Comment by RuchitaCrawl** [comment]
Thanks for your reply :)

 yaa it's all the plugins have same version latest 19.

```
   " @ckeditor/ckeditor5-basic-styles": "^19.0.0",
    "@ckeditor/ckeditor5-build-classic": "^19.0.0",
    "@ckeditor/ckeditor5-build-decoupled-document": "^19.0.0",
    "@ckeditor/ckeditor5-editor-classic": "^19.0.0",
    "@ckeditor/ckeditor5-editor-decoupled": "^19.0.0",
    "@ckeditor/ckeditor5-image": "^19.0.0",
    "@ckeditor/ckeditor5-link": "^19.0.0",
```

Which information you need?

**Comment by FilipTokarski** [comment]
Oh, sorry, I didn't notice that at the very top you are importing an already built editor and then adding a plugin. That apparently causes the problem. If you want to use a custom editor in Vue, please refer to [our 'building from source in Vue' docs](https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/frameworks/vuejs.html#using-ckeditor-from-source). That should solve the problem.

**Comment by RuchitaCrawl** [comment]
Thanks For your valuable answer :)

I also tried using source . but it don't show me tooltip.

 ```
   import ClassicEditor from '@ckeditor/ckeditor5-editor-classic/src/classiceditor';
    import ImageResize from '@ckeditor/ckeditor5-image/src/imageresize';
    import Bold from '@ckeditor/ckeditor5-basic-styles/src/bold';
    import Italic from '@ckeditor/ckeditor5-basic-styles/src/italic';
    import Underline from '@ckeditor/ckeditor5-basic-styles/src/underline';
    import Strikethrough from '@ckeditor/ckeditor5-basic-styles/src/strikethrough';
    import Code from '@ckeditor/ckeditor5-basic-styles/src/code';
    import Subscript from '@ckeditor/ckeditor5-basic-styles/src/subscript';
    import Superscript from '@ckeditor/ckeditor5-basic-styles/src/superscript';

data() {
            return {
                editor: ClassicEditor,
                editorData: '',
                plugins: [Bold, Italic, Underline, Strikethrough, Code, Subscript, Superscript, ImageResize ],
                editorConfig: {
                    ckfinder: {
                        // eslint-disable-next-line max-len
                        uploadUrl: 'https://cksource.com/weuy2g4ryt278ywiue/core/connector/php/connector.php?command=QuickUpload&type=Files&responseType=json'
                    },
                    toolbar: {
                        items: [
                            'bold', 'italic', 'underline', 'strikethrough', 'code','subscript', 'superscript'
                        ]
                    },
                    image: {
                        // toolbar: [ 'imageTextAlternative', '|', 'imageStyle:alignLeft', 'imageStyle:full', 'imageStyle:alignRight'],

                        styles: [
                            'full',
                            'alignLeft',
                            'alignRight',
                        ],
                    },
                },
            }
        },
```

It is showing [empty](https://prnt.sc/snzcea) space.

**Comment by FilipTokarski** [comment]
Hi, sorry for late reply. A you sure you added `:config` to your component?
```js
<template>
    <div id="app">
        <ckeditor :editor="editor" v-model="editorData" :config="editorConfig"></ckeditor>
    </div>
</template>
```

**Comment by RuchitaCrawl** [comment]
yes i already added this..

 ```
<div class="">
     <ckeditor :editor="editor" v-model="form.description" :config="editorConfig" @ready="onReady"> 
     </ckeditor>
</div>
```

**Comment by FilipTokarski** [comment]
That's strange, I cannot reproduce this missing toolbar. Do you get any console errors or warnings?

**Comment by RuchitaCrawl** [comment]
Yes, it give me warning  " toolbarview-item-unavailable: The requested toolbar item is unavailable".  for all above plugin. even if i installed it already.

**Comment by FilipTokarski** [comment]
Ok, I think I found the problem. 
First, as it comes to `Paragraph` plugin, it is needed so that you can type in the editor, it enables basic editing.  
  
As it comes to your configuration, here's what I found:
- `plugins` list is outside the `editorConfig`, which results in editor not recognising imported plugins and trying to add toolbar items to missing plugins. So please add it to configuration:
```js
editorConfig: {
             plugins: [ EssentialsPlugin, Bold, Italic, Underline, Strikethrough, Code, Subscript, Superscript, Image, ImageResize ],
             toolbar: {
                  items: [
                      'bold', 'italic', 'underline', 'strikethrough', 'code','subscript', 'superscript'
                  ]
              },
             image: {
                  styles: [
                      'full',
                      'alignLeft',
                      'alignRight',
                  ],
              },
          },
```
- you should also import `Image` plugin and add it to the list. Without it, you will most likely be facing and error when trying to configure image properties, so in your imports please do:
```js
import Image from '@ckeditor/ckeditor5-image/src/image';
import ImageResize from '@ckeditor/ckeditor5-image/src/imageresize';
```

Please, let me know if this helps.

**Comment by RuchitaCrawl** [comment]
I refer your steps but paragraph plugin give me `duplicate module run time error`.

If i not import it then it works good bt give me `"TypeError: Cannot read property 'getAttribute' of null"` .   then i refer github issues but it don't work for me.

Can you say that how can i implement image resize handle in decoupled decument editor buid?

**Comment by FilipTokarski** [comment]
Hi, it's hard for me to guess now what the problem is, because the code you send works fine on my machine. Also few days ago we had a release ( 19.1.0 ) and now I'm not sure if you are still using the same packages, or mixing different versions. Without a detailed step-by-step info of what you do and how you integrate CKEditor with Vue, there's not much I can add now. Maybe you could just once again try to [create a fresh integration from source](https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/frameworks/vuejs.html#using-ckeditor-from-source) or [download a custom build](https://ckeditor.com/ckeditor-5/online-builder/) and add it to the Vue app.

**Comment by RuchitaCrawl** [comment]
Ok, i will give you step-by-step detailed info.

My packages is :: 

   ```
 "@ckeditor/ckeditor5-basic-styles": "^19.0.1",
    "@ckeditor/ckeditor5-build-classic": "^19.0.0",
    "@ckeditor/ckeditor5-build-decoupled-document": "^19.0.0",
    "@ckeditor/ckeditor5-ckfinder": "^19.0.0",
    "@ckeditor/ckeditor5-dev-utils": "^20.0.0",
    "@ckeditor/ckeditor5-dev-webpack-plugin": "^20.0.0",
    "@ckeditor/ckeditor5-editor-classic": "^19.0.1",
    "@ckeditor/ckeditor5-editor-decoupled": "^19.0.0",
    "@ckeditor/ckeditor5-essentials": "^19.0.1",
    "@ckeditor/ckeditor5-image": "^19.0.0",
    "@ckeditor/ckeditor5-link": "^19.0.1",
    "@ckeditor/ckeditor5-theme-lark": "^19.1.0",
    "@ckeditor/ckeditor5-ui": "^19.0.1",
    "@ckeditor/ckeditor5-vue": "^1.0.1",
    "@ckeditor/ckeditor5-widget": "^19.0.0",
    "child_process": "^1.0.2",
    "fs": "0.0.1-security",
```

I have created vue.config.js in root of my project directory.

```
const path = require( 'path' );
const CKEditorWebpackPlugin = require( '@ckeditor/ckeditor5-dev-webpack-plugin' );
const { styles } = require( '@ckeditor/ckeditor5-dev-utils' );

module.exports = {
    // The source of CKEditor is encapsulated in ES6 modules. By default, the code
    // from the node_modules directory is not transpiled, so you must explicitly tell
    // the CLI tools to transpile JavaScript files in all ckeditor5-* modules.
    transpileDependencies: [
        /ckeditor5-[^/\\]+[/\\]src[/\\].+\.js$/,
    ],

    configureWebpack: {
        plugins: [
            // CKEditor needs its own plugin to be built using webpack.
            new CKEditorWebpackPlugin( {
                // See https://ckeditor.com/docs/ckeditor5/latest/features/ui-language.html
                language: 'en'
            } )
        ]
    },

    // Vue CLI would normally use its own loader to load .svg and .css files, however:
    //	1. The icons used by CKEditor must be loaded using raw-loader,
    //	2. The CSS used by CKEditor must be transpiled using PostCSS to load properly.
    chainWebpack: config => {
        // (1.) To handle editor icons, get the default rule for *.svg files first:
        const svgRule = config.module.rule( 'svg' );

        // Then you can either:
        //
        // * clear all loaders for existing 'svg' rule:
        //
        //		svgRule.uses.clear();
        //
        // * or exclude ckeditor directory from node_modules:
        svgRule.exclude.add( path.join( __dirname, 'node_modules', '@ckeditor' ) );

        // Add an entry for *.svg files belonging to CKEditor. You can either:
        //
        // * modify the existing 'svg' rule:
        //
        //		svgRule.use( 'raw-loader' ).loader( 'raw-loader' );
        //
        // * or add a new one:
        config.module
            .rule( 'cke-svg' )
            .test( /ckeditor5-[^/\\]+[/\\]theme[/\\]icons[/\\][^/\\]+\.svg$/ )
            .use( 'raw-loader' )
            .loader( 'raw-loader' );

        // (2.) Transpile the .css files imported by the editor using PostCSS.
        // Make sure only the CSS belonging to ckeditor5-* packages is processed this way.
        config.module
            .rule( 'cke-css' )
            .test( /ckeditor5-[^/\\]+[/\\].+\.css$/ )
            .use( 'postcss-loader' )
            .loader( 'postcss-loader' )
            .tap( () => {
                return styles.getPostCssConfig( {
                    themeImporter: {
                        themePath: require.resolve( '@ckeditor/ckeditor5-theme-lark' ),
                    },
                    minify: true
                } );
            } );
    }
};
```
In my vue component

`<ckeditor :editor="editor" v-model="form.description" :config="editorConfig" @ready="onReady"></ckeditor>`
```

    import CKEditor from '@ckeditor/ckeditor5-vue';
    import DecoupledEditor from '@ckeditor/ckeditor5-editor-decoupled/src/decouplededitor';
    import Essentials from '@ckeditor/ckeditor5-essentials/src/essentials';
    import Paragraph from '@ckeditor/ckeditor5-paragraph/src/paragraph';
    import Bold from '@ckeditor/ckeditor5-basic-styles/src/bold';
    import Italic from '@ckeditor/ckeditor5-basic-styles/src/italic';
    import Underline from '@ckeditor/ckeditor5-basic-styles/src/underline';
    import Strikethrough from '@ckeditor/ckeditor5-basic-styles/src/strikethrough';
    import Code from '@ckeditor/ckeditor5-basic-styles/src/code';
    import Subscript from '@ckeditor/ckeditor5-basic-styles/src/subscript';
    import Superscript from '@ckeditor/ckeditor5-basic-styles/src/superscript';
    import Image from '@ckeditor/ckeditor5-image/src/image';
    import ImageResize from '@ckeditor/ckeditor5-image/src/imageresize';

 export default {
        components:{
            ckeditor: CKEditor.component,
        },
        data() {
            return {
                editor: DecoupledEditor,
                editorData: '',
                editorConfig: {
                     plugins: [Essentials, Bold, Italic, Underline,Paragraph, Strikethrough, Code, Subscript, Superscript, Image, ImageResize],
                    ckfinder: {
                        // eslint-disable-next-line max-len
                        uploadUrl: 'https://cksource.com/weuy2g4ryt278ywiue/core/connector/php/connector.php?command=QuickUpload&type=Files&responseType=json'
                    },
                     toolbar: {
                         items: [
                            'essentials', 'bold', 'italic', 'underline', 'paragraph', 'strikethrough', 'code','subscript', 'superscript', image
                         ]
                     },
                    image: {
                         toolbar: [ 'imageTextAlternative', '|', 'imageStyle:alignLeft', 'imageStyle:full', 'imageStyle:alignRight'],

                        styles: [
                            'full',
                            'alignLeft',
                            'alignRight',
                        ],
                    },
                },
            }
        },
methods: {
 onReady( editor )  {
                // Insert the toolbar before the editable area.
                editor.ui.getEditableElement().parentElement.insertBefore(
                    editor.ui.view.toolbar.element,
                    editor.ui.getEditableElement()
                );
            },
}
```

Now can you say what's the issue?

**Comment by FilipTokarski** [comment]
I was able to run this code with no errors, only things I had to modify:
1. You don't have Paragraph plugin installed and you import it, which causes error.
2. You don't need to add `'paragraph'` to toolbar items.
3. You use `CKFinder` configuration, but have no `CKFinder` installed and imported.
4. You have `imageStyles` and `imageToolbar` configuration but no [ImageStyles and ImageToolbar plugins imported](https://ckeditor.com/docs/ckeditor5/latest/features/image.html#installation).
  
Apart from this I have no idea what is `form.description` in your `v-model` and that as well might be source of the problem.
Having said that, I guess there is some problem on your side and not with CKEditor, so I advise to double check everything.
